### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Ananyamadhu08**
- added authentication functionality.

### CODE REVIEW

File: src/components/Cart-page/AddressBox.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        </h6>
+        <p className="text-md text-rose-200">
+          sddsjnc jsdsn sdsdkns ds sddsjnc jsdsn sdsdkns ds sddsjnc jsdsn
+          sdsdkns ds sddsjnc jsdsn sdsdkns ds sddsjnc jsdsn sdsdkns ds sddsjnc
+        </p>
+      </div>
+
+      <button
+        className="bg-rose-300 btn-sm text-slate-900 "
+        style={{ left: "2rem", bottom: "2rem" }}
```
Comment (anshulraheja) [inline src/components/Cart-page/AddressBox.jsx:21]: try to avoid inline style

**Review by anshulraheja — APPROVED** [review]

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=19387
```
Comment (VedanthB) [inline .env:1]: dont commit .env file, it has secret info , add .env in a .gitignore file to ignore the file during commit

File: .gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
-node_modules/
\ No newline at end of file
+node_modules/
+.env/
```
Comment (VedanthB) [inline .gitignore:2]: .env not .env/

File: src/components/global/Header/HeaderRight.jsx [hunk 4]
```diff
@@ -1,19 +1,35 @@
 import React from "react";
 import { Link } from "react-router-dom";
+import { authActions } from "../../../contexts/constants/authConstants";
```
Comment (VedanthB) [inline src/components/global/Header/HeaderRight.jsx:3]: you can do a barrel import / export

File: src/contexts/reducers/authReducer.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { authActions } from "../constants/authConstants";
+
+// encodedToken: null,
```
Comment (VedanthB) [inline src/contexts/reducers/authReducer.js:3]: remove commented code

**Review by VedanthB — APPROVED** [review]
Good Work Ananya, 

I have suggested a few changes, please look into those before merging the PR 🚀

**Pull request merged**